src/decision_tree.cpp

```diff
@@ -3,13 +3,17 @@
 #include "datasets.hpp"
 #include "losses.hpp"
 #include <algorithm>  // std::sort.
+#include <stack>  // std::stack.
 #include <assert.h>
 #include <math.h>  // std::sqrt.
 #include <time.h>  // std::time.
 
 // Constructors:
 
-DecisionTree::DecisionTree(DataFrame dataframe, bool regression, std::string loss, int mtry, int max_height, int max_leaves)
```

Comment:
A vanilla DecisionTree doesn't need an `mtry` parameter, as it will always try all features as splits.

It might also be better to make separate Classifier and Regressor versions of DecisionTree, as their `fit` and `predict` will differ.